### PR TITLE
Feature/hakodate

### PR DESCRIPTION
**Pull request by tkyonezu**
## What is this pull request?

Build IROHA with makefile. You can build IROHA using make target and create and execute a docker image. Currently, the following target is prepared.

```
# - all (default) - buid iroha-build container, and build iroha
# - docker - build iroha runtime container
# - docker-test - build iroha runtime and test container
# - up - running iroha container by docker-compose
# - down - stop and remove iroha container by docker-compose
# - test-up - running iroha-test container buy docker-compose
# - test - down - stop and remove iroha - test container by docker - compose
# - test - exec all test commands
# - clean - cleaning protobuf schemas and build directory
```

In the target test, the commands created in test_bin are executed in alphabetical order.
  
## Why do you implement it? Why do we need this pull request?

It is to make IROHA easy to build and run for testing.
  
## How to use the features provided in the pull request?

With make alone, create an iroha-docker-develop container, build iroha, and create a docker image of IROHA for execution.
By executing make up, IROHA's docker container is started using docker-compose. You can stop and delete the docker container of IROHA started up by make down.

## Details/Features

Add Makefile and some scripts under scripts directory.

### CODE REVIEW

**Comment by tkyonezu** [comment]
May be this pull request is better for develop branch.

File: docker/develop/Dockerfile [hunk 1]
```diff
@@ -142,10 +142,10 @@ RUN git clone https://github.com/01org/tbb /tmp/tbb-2017_U7; \
     rm -rf /tmp/tbb-2017_U7
 
 # install docker
-RUN curl -L -o /tmp/docker-17.06.0-ce.tgz https://download.docker.com/linux/static/stable/x86_64/docker-17.06.0-ce.tgz; \
-    tar -xz -C /tmp -f /tmp/docker-17.06.0-ce.tgz; \
+RUN curl -L -o /tmp/docker-17.06.2-ce.tgz https://download.docker.com/linux/static/stable/x86_64/docker-17.06.2-ce.tgz; \
```
Comment (Warchant) [inline docker/develop/Dockerfile]: Is there something special in the version 17.06.02-ce that does not exist in 17.06.0-ce?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+up: iroha-up
+
+test-up: iroha-test-up
+
+down: iroha-down
+
+test-down: iroha-test-down
+
+iroha-develop:
+	cd docker/develop && docker build --rm  -t hyperledger/iroha-docker-develop .
```
Comment (Warchant) [inline Makefile]: you can add 
```
--build-args PARALLELISM=8
```
for faster builds :)

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -2,25 +2,24 @@ version: '3'
 
 services:
   node:
-    image: hyperledger/iroha-docker-develop
+    image: hyperledger/iroha
     ports:
       - "50051:50051"
     environment:
-      - IROHA_POSTGRES_HOST=${COMPOSE_PROJECT_NAME}_postgres_1
+      - IROHA_POSTGRES_HOST=172.18.0.3
```
Comment (Warchant) [inline docker/docker-compose.yml]: it is very bad to hardcode IPs in docker-compose files. IP can be changed any time. It is better to use name of the service.

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -2,25 +2,24 @@ version: '3'
...
       - IROHA_POSTGRES_PORT=5432
       - IROHA_POSTGRES_USER=iroha
-      - IROHA_POSTGRES_PASSWORD=helloworld
-      - IROHA_REDIS_HOST=${COMPOSE_PROJECT_NAME}_redis_1
+      - IROHA_POSTGRES_PASSWORD=HelloW0rld
+      - IROHA_REDIS_HOST=172.18.0.2
```
Comment (Warchant) [inline docker/docker-compose.yml]: Same as above here.

File: docker/release/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+COPY iroha ${IROHA_HOME}
+
+RUN apt update && \
+    apt upgrade -y && \
+    useradd -ms /bin/bash iroha && \
+    chown -R iroha:iroha ${IROHA_HOME}
+
+WORKDIR ${IROHA_HOME}
+
+CMD sleep 20; ${IROHA_HOME}/bin/irohad --config config/iroha.conf --genesis_block config/zero.block
```
Comment (Warchant) [inline docker/release/Dockerfile]: Why `sleep 20`?

File: scripts/iroha-test.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+LIST=$(docker exec -t iroha_node_1 /bin/ls /usr/local/iroha/test_bin | sed 's/\//')
```
Comment (Warchant) [inline scripts/iroha-test.sh]: What is this weird symbol? `^M`.

**Comment by neewy** [comment]
Thank you for your contribution, Yonezu-san.
Please, can you correct hardcoded values in Makefile?

**Comment by neewy** [comment]
@tkyonezu there is no activity for 20 days, regarding the suggested changes of reviewer (@Warchant). According to Iroha working agreement 4.10.4 https://github.com/hyperledger/iroha/wiki/Iroha-working-agreement the person, who opened pull request should clarify whether review fixes will be implemented or not in period of 24 hours.

**Comment by tkyonezu** [comment]
@neewy I already closed this PR and this is superseeded by PR #605, as I said to you before. 
And I've rewite IP address to hostname on PR #605.  Thank you for your comment.

**Pull request closed without merging**